### ISSUE TITLE
Audiosurf-RPC Activity stays on "Loading" on Discord

### ISSUE THREAD

**Issue by dizzydoo** [RubberDuckShobe/audiosurf-rpc#2]
Hi, I'm running the latest release version of audiosurf-rpc but the rich presence on Discord fails to update when choosing a mode or while playing a song. The  dll clearly is capturing the song names that Audiosurf sees. The Rich Presence simply fails to switch to a mode that allows it to display song info. 

![image](https://user-images.githubusercontent.com/17954168/127795527-fbb910b3-d857-4beb-b830-e8d4601529d6.png)

If it's of any use to debug this, the only way I could get audiosurf-rpc to work was by copying all the files in the release zip to the channels folder inside of the Audiosurf folder (or `C:\Program Files (x86)\Steam\steamapps\common\Audiosurf\engine\channels`) and then creating a second channels folder inside of there with the Crypt.dll included in the zip file.

![image](https://user-images.githubusercontent.com/17954168/127796068-454bc30b-4213-4897-b097-85d8e4cab4de.png)

**Comment by RubberDuckShobe** [comment]
Thank you for the report!

I'm not sure what could be causing this, but I'll take a look at it.

**Comment by RubberDuckShobe** [comment]
Actually, from your screenshot I've noticed it doesn't seem to capture the WM_COPYDATA messages Audiosurf sends to signal what state it is in. I'll try to investigate this.

**Comment by RubberDuckShobe** [comment]
Can you try using the dll from this .zip file? It should attempt re-registering the window if it detects a FileRef without being registered to receive WM_COPYDATA messages.

[audiosurf-rpc-test.zip](https://github.com/AudiosurfResearch/audiosurf-rpc/files/6915870/audiosurf-rpc-test.zip)

**Comment by dizzydoo** [comment]
Hi, I've replaced the dll in my game folder with the one you've attached, and the output is updated to say the window isn't registered, but doesn't seem like it can re-register. The activity still says "Loading" even if I'm in the middle of a track. Same result whether launching through steam or directly from the game exe while running as admin.

![image](https://user-images.githubusercontent.com/17954168/127872321-b8da71d3-876a-4ab3-861f-8b8317987f31.png)

Also just as an update to me mentioning how I installed the mod, as a sanity check I tried installing it as listed in the readme and it failed to load properly. Putting all four main dlls in the channels folder allows it to run, meaning that me copying the channels folder with the Crypt.dll file inside was unnecessary.
